### DISCUSSION TITLE
Is there any way to configure security key via environment variable?

### DISCUSSION THREAD

**Discussion by andrew-demb** [NoiseByNorthwest/php-spx#203] [Q&A] [answered]
We are using one docker image for running applications in different environments and we want to change the required security key for different environments (for example, for stage/production)

Is there any way to configure `spx.http_key` with environment variable on start a `php-fpm` process?

**Comment by NoiseByNorthwest** [comment]
No it is currently an INI entry only. You can however override your php-fpm container's entry point in order to add `-d spx.http_key=<your secret>` to its argument list.

**Comment by andrew-demb** [comment, marked as answer]
We found a way to use environment variables in PHP docs and just checked in production - it works like a charm.

https://www.php.net/manual/en/configuration.file.php

```ini
spx.http_key="${PHP_SPX_SECRET}"
```
